### PR TITLE
Dual (or more!) Wielding

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
🆑 Kor
rscadd: You can once again dual wield guns. Firing a gun will automatically fire the gun held in your off hand if you are on harm intent. This will impose a large accuracy penalty.
rscadd: If you happen to have more than two arms, you can fire as many guns as you can hold. The accuracy penalty will stack for each gun though.
rscadd: Weapons that previously had extra recoil and a chance to drop now instead require two weapons to fire, as the previous mechanic tended to just confuse people.
/🆑

### CODE REVIEW

**Comment by YakumoChen** [comment]
Wasn't this hilariously abused by Mekhi last time it got tested?

**Comment by KorPhaeron** [comment]
@YakumoChen 

That's why it has massive accuracy penalties now

File: code/modules/projectiles/gun.dm [hunk 1]
```diff
@@ -170,7 +170,22 @@
...
+	var/loop_counter = 0
+	if(ishuman(user) && user.a_intent == "harm")
+		var/mob/living/carbon/human/H = user
+		for(var/obj/item/weapon/gun/G in H.held_items)
+			if(G == src)
+				continue
+			else if(G.can_trigger_gun(user))
+				bonus_spread += 24 * G.weapon_weight
+				loop_counter++
+				spawn(loop_counter)
```
Comment (JJRcop) [inline code/modules/projectiles/gun.dm]: wow using the old spawn proc what is this 2015

**Comment by Iamgoofball** [comment]
@KorPhaeron if you're named max payne it should give no accuracy penalties

also if you turn your guns sideways you should also get lessened accuracy penalties

**Comment by PKPenguin321** [comment]
Unless the shots can go fully to your sides or behind you with the accuracy penalty, all it will do is effectively turn your guns into spreadshot

**Comment by Shadowlight213** [comment]
Tase and laze bby!

**Comment by Dorsisdwarf** [comment]
Spread isnt necessarily a bad thing, especially when multi-firing

**Comment by KorPhaeron** [comment]
If the spread is too much or too little it's easy to adjust

**Comment by WJohn** [comment]
Last time spread was 12 degrees deviation, now it looks to be 24, so it should be pretty intense.

Of course, for combat shotguns with buckshot you still do a ludicrous amount of damage. Pretty sure those only require one hand to hold.

**Comment by GunHog** [comment]
Time to wear a formal suit, nitrile gloves, a clown mask, and hold two silenced Syndie pistols! Payday 2 Akimbo style!

**Comment by Supermichael777** [comment]
i cant hit shit anyway soooooo

**Comment by Dorsisdwarf** [comment]
This is definitely something to be testmerged though

File: code/modules/projectiles/gun.dm [hunk 2]
```diff
@@ -170,7 +170,22 @@
...
+					G.recharge_newshot()
```
Comment (phil235) [inline code/modules/projectiles/gun.dm]: why is this here?

**Comment by ChangelingRain** [comment]
Now for the real question

## Does Accuracy Matter When You're Shooting A Blob 7 Tiles Wide?

**Comment by Bawhoppen** [comment]
Mem

**Comment by kevinz000** [comment]
Only if your accuracy is less than 7 tiles

File: code/modules/projectiles/gun.dm (hunk 2 above)
Comment (KorPhaeron) [inline code/modules/projectiles/gun.dm]: I honestly don't exactly remember since I wrote this months ago, but I do remember guns wouldn't process the chamber/reload properly when you were forcing them to fire offhand rather than going through all the steps of an actual click/attack, so forcing it to reload was needed.

**Comment by KorPhaeron** [comment]
Man, blob mode always ruins all the fun

**Comment by Tacolizard** [comment]
gun size weight should affect accuracy penalty. shotguns and laser cannons should spread massively, but not tasers or stetchkins

**Comment by KorPhaeron** [comment]
That is already the case

**Comment by WJohn** [comment]
What about revolvers? Those are at least small size and those do ludicrous amounts of damage per shot.

**Comment by KorPhaeron** [comment]
@WJohn 

Those would probably be pretty strong, but 26 TC is out of reach for anyone but nuke ops anyway

**Comment by kevinz000** [comment]
Dual wield KAs.

**Comment by jughu** [comment]
Who needs accuracy if you can just spray buckshots everywhere

**Comment by Lexorion** [comment]
Nice!

**Comment by Dorsisdwarf** [comment]
I wanna see someone with a nodrop implant, six arms, and six L6 SAWs go to town

File: code/modules/projectiles/gun.dm (hunk 2 above)
Comment (phil235) [inline code/modules/projectiles/gun.dm]: I fixed the underlying issue that was also affecting burst fire. You don't need that line anymore.

**Comment by PKPenguin321** [comment]
>I wanna see someone with a nodrop implant, six arms, and six L6 SAWs go to town

We bullet hell now

**Comment by jughu** [comment]
>wavedashes

**Comment by kevinz000** [comment]
https://www.youtube.com/watch?v=HSsz2tJl8rg&feature=youtu.be&t=14s

**Comment by KorPhaeron** [comment]
@optimumtact 

Shotguns can no longer be dual weilded
